matlab_simulation/08-planning/lib/wavefront.m

```diff
@@ -0,0 +1,112 @@
+function [wavefrontmap, path] = wavefront(curmap, endpos, startpos)
+%input: binary arrary representing the environment, 0 is object, anything greater than 0 is open space
+%input: start position of the robot
+%input: end position or goal location
+
+%output: wave - array with with BFS values
+%output: path - vector of cells or points the robot should pass through 
+
+%use a java linked list as an effiecient queue
+import java.util.LinkedList
+queue = LinkedList();
+
+%add a 0 padding around the current map to ensure the environment is closed.
+%also means you do not need to check for boundary conditions. 
+curmap = padarray(curmap,1,0);
+
+% make an output map that is initilized to all -10, which encodes unexplored cells;
+wavefrontmap = -10*ones(size(curmap));
+
+%adjust the value of endpos to agree with the new padding. 
+endpos = endpos + [1,1];
+
```

Comment:
Do we need to adjust the value for startpos as well to accommodate for the padding?